### DISCUSSION TITLE
ItemsControl - Display placeholder when item source is empty

### DISCUSSION THREAD

**Discussion by JaggerJo** [AvaloniaUI/Avalonia#14807] [Q&A]
When using an items control but the item source does not contain any items I usually want to inform the user on how to add items. 

Example: Playlist is empty. Click here to add songs.

I could use: 
- a panel that contains the ItemsControl and the Empty Placeholder and conditionally set their visibility.
- a custom Template for the items control that already has that functionality baked in.
- a custom control where you can supply a `EmptyContent` property.

I feel like this is a common use case. Am I missing something existing here?

**Comment by timunie** [comment]
No one-fits-all here, so you need to pick any of your ideas I guess

**Reply by JaggerJo** [reply]
Thanks for the fast response Tim! Ok, will go with a custom control then.
